src/Umbraco.Core/Migrations/Upgrade/v_8_8_0/TagControlChangeDataColumn.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Linq;
+using Umbraco.Core.Migrations.Upgrade.V_8_0_0.Models;
+using Umbraco.Core.Models;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+
+namespace Umbraco.Core.Migrations.Upgrade.V_8_8_0
+{
+    public class TagControlChangeDataColumn : MigrationBase
+    {
+        public TagControlChangeDataColumn(IMigrationContext context)
+            : base(context)
+        {
+
+        }
+
+        /// <summary>
+        /// Changes the configured storage column <c>dbType</c> in <c>umbracoDataType</c> to nText where the <c>Umbraco.Tags</c> property editor is used.
+        /// Migrates data in <c>umbracoPropertyData</c> from varcharValue to textValue for affected properties.
+        /// </summary>
+        public override void Migrate()
+        {
+            //get UmbracoTags datatypes that need to be migrated
+            var sqlDataTypes = Sql()
+               .Select<DataTypeDto>()
+               .From<DataTypeDto>()
+               .Where<DataTypeDto>(x => x.EditorAlias == Constants.PropertyEditors.Aliases.Tags);
+            var dataTypes = Database.Fetch<DataTypeDto>(sqlDataTypes).ToList();
+
+            foreach (var dataType in dataTypes)
```

Comment:
I completely agree. My previous iteration of this pull request used raw SQL to do bulk updates as you have shown (and I tested this as working) but @nul800sebastiaan requested that it was changed to fluent style and I couldn't bulk update without the loop. See https://github.com/umbraco/Umbraco-CMS/pull/8469/commits/b046bad6cc954aef5e978d4215f0fdde880a1cd4

Reply:
Ah i see, i think he was probably referring to making the queries strongly typed so there's no mistakes including casing, etc... with raw sql code or if we change things.

Reply:
For example:

```cs
Update
    .Table(Constants.DatabaseSchema.Tables.DataType)
    .Set(new { dbType = "Ntext" })
    .Where(new { propertyEditorAlias = Constants.PropertyEditors.Aliases.Tags })
    .Do();

// OR

var updateSql1 = Context.SqlContext.Sql()
    .Update<DataTypeDto>(x => x.Set(p => p.DbType, ValueStorageType.Ntext.ToString()))
    .Where<DataTypeDto>(x => x.EditorAlias == Constants.PropertyEditors.Aliases.Tags);
Context.Database.Execute(updateSql1);
```

For the other one its a bit more tricky because I don't think we have an Update syntax that can set to another column. Here's what it could look like if we could do that:

```cs
var subQuerySql2 = Context.SqlContext.Sql().Select<DataTypeDto>(x => x.NodeId).From<DataTypeDto>()
    .Where<DataTypeDto>(x => x.EditorAlias == Constants.PropertyEditors.Aliases.Tags);

var subQuerySql1 = Context.SqlContext.Sql().Select<PropertyTypeDto>(x => x.Id).From<PropertyTypeDto>()
    .WhereIn<PropertyTypeDto>(x => x.DataTypeId, subQuerySql2);                

var updateSql2 = Context.SqlContext.Sql()
    .Update<PropertyDataDto>(x => x.Set(p => p.TextValue, "TODO: WE DON'T HAVE THIS FUNCTIONALITY"))
    .WhereIn<PropertyTypeDto>(x => x.Id, subQuerySql1)
    .Where<PropertyDataDto>(x => x.TextValue == null)
    .Where<PropertyDataDto>(x => x.VarcharValue != null);

Context.Database.Execute(updateSql2);
```

It's probably possible to update the `SqlUpd<TDto>` class and functionality to support passing in another Expression for the source column.  But in the meantime this should be possible

```cs
var subQuerySql2 = Context.SqlContext.Sql().Select<DataTypeDto>(x => x.NodeId).From<DataTypeDto>()
    .Where<DataTypeDto>(x => x.EditorAlias == Constants.PropertyEditors.Aliases.Tags);

var subQuerySql1 = Context.SqlContext.Sql().Select<PropertyTypeDto>(x => x.Id).From<PropertyTypeDto>()
    .WhereIn<PropertyTypeDto>(x => x.DataTypeId, subQuerySql2);

var updateSql2 = Context.SqlContext.Sql()
    .Update<PropertyDataDto>()
    .Append("SET textValue = varcharValue, varcharValue = null")
    .WhereIn<PropertyTypeDto>(x => x.Id, subQuerySql1)
    .Where<PropertyDataDto>(x => x.TextValue == null)
    .Where<PropertyDataDto>(x => x.VarcharValue != null);

Context.Database.Execute(updateSql2);
```

Reply:
I was under the impression that doing a direct SQL query would prevent us from rolling it back if something in the upgrade goes wrong somewhere, I might have been mistaken there! 

Also we need to make absolutely sure that the query works on SQL CE as well then.

Reply:
Migrations execute in a single transaction. Each migration fluent script just executes SQL behind the scenes. We already do custom SQL queries in some migrations where the fluent script does not cater for our needs. And of course, the query needs to work in all supported DBs :)